src/test/java/com/artipie/npm/NpmITCase.java

```diff
@@ -114,6 +117,41 @@ void npmInstall() throws Exception {
         );
     }
 
+    @Test
+    void npmPublish() throws Exception {
+        final boolean anonymous = true;
+        final String proj = "@hello/simple-npm-project";
+        final Key path = new Key.From("repos/my-npm");
+        this.init(anonymous);
+        new TestResource("npm/simple-npm-project")
+            .addFilesTo(this.storage, new Key.From(proj));
+        MatcherAssert.assertThat(
+            "Package was published",
+            this.exec("npm", "publish", proj, "--registry", this.url),
+            new StringContains("+ @hello/simple-npm-project@1.0.1")
+        );
+        final JsonObject meta = new JsonFromPublisher(
+            this.storage.value(
+                new Key.From(path, proj, "meta.json")
+            ).toCompletableFuture().join()
+        ).json().toCompletableFuture().join();
+        MatcherAssert.assertThat(
+            "Metadata should be valid",
+            meta.getJsonObject("versions")
+                .getJsonObject("1.0.1")
+                .getJsonObject("dist")
+                .getString("tarball"),
+            new IsEqual<>(String.format("/%s/-/%s-1.0.1.tgz", proj, proj))
+        );
+        MatcherAssert.assertThat(
+            "File should be in storage after publishing",
+            this.storage.exists(
+                new Key.From(path, String.format("%s/-/%s-1.0.1.tgz", proj, proj))
+            ).toCompletableFuture().join(),
```

Comment:
@genryxy `.tgz` file path is constructed twice here, I'd suggest to extract into variable